src/rpc/net.cpp

```diff
@@ -943,7 +943,7 @@ static RPCHelpMan addpeeraddress()
     }
 
     const std::string& addr_string{request.params[0].get_str()};
-    const uint16_t port{static_cast<uint16_t>(request.params[1].get_int())};
+    const uint16_t& port{request.params[1].get_uint16()};
```

Comment:
```suggestion
    const auto port{request.params[1].getInt<uint16_t>()};
```

on master this is now trivial to implement in a one-line diff